web/assets/js/actions.ts

```diff
@@ -1,214 +1,391 @@
+
+// TODO move this out into an ENV variable that's loaded in
+const wsURL: string = "ws://127.0.0.1:26657/websocket"
+const chessRealm: string = "gno.land/r/gnochess"
+
+/**
+ * Actions is a singleton logic bundler
+ * that is shared throughout the game.
+ *
+ * Always use as Actions.getInstance()
+ */
+// @ts-ignore
+class Actions {
+    private static instance: Actions;
+
+    private wallet: GnoWallet | null = null;
+    private provider: GnoWSProvider | null = null;
+    private faucetToken: string | null = null;
+
+    private constructor() {
+    }
+
+    /**
+     * Fetches the Actions instance. If no instance is
+     * initialized, it initializes it
+     */
+    public static async getInstance(): Promise<Actions> {
+        if (!Actions.instance) {
+            Actions.instance = new Actions()
+
+            await Actions.instance.initialize()
+        }
+
+        return Actions.instance
+    }
+
+    /**
+     * Prepares the Actions instance
+     * @private
+     */
+    private async initialize() {
+        // Wallet initialization //
+
+        // Try to load the mnemonic from local storage
+        let mnemonic: string | null = localStorage.getItem(defaultMnemonicKey)
+        if (!mnemonic || mnemonic === "") {
+            // Generate a fresh mnemonic
+            mnemonic = generateMnemonic()
+
+            // Save the mnemonic to local storage
+            saveToLocalStorage(defaultMnemonicKey, mnemonic)
+        }
+
+        // Initialize the wallet using the saved mnemonic
+        this.wallet = await GnoWallet.fromMnemonic(mnemonic);
+
+        // Initialize the provider
+        this.provider = new GnoWSProvider(wsURL)
+
+        // Connect the wallet to the provider
+        this.wallet.connect(this.provider)
+
+        // Faucet token initialization //
+        let faucetToken: string | null = localStorage.getItem(defaultFaucetTokenKey)
+        if (faucetToken && faucetToken !== "") {
+            // Faucet token initialized
+            this.faucetToken = faucetToken
+        }
+    }
+
+    /**
+     * Saves the faucet token to local storage
+     * @param token the faucet token
+     */
+    public setFaucetToken(token: string) {
+        this.faucetToken = token
+
+        localStorage.setItem(defaultFaucetTokenKey, token)
+    }
+
+    /**
+     * Fetches the saved faucet token, if any
+     */
+    public getFaucetToken(): string | null {
+        return this.faucetToken
+    }
+
+    /****************
+     * GAME ENGINE
+     ****************/
+
+    /**
+     * Joins the waiting lobby for the game
+     * @param time
+     */
+    public async joinLobby(time: GameTime): Promise<GameSettings> {
+        // Join the waiting lobby
+        const joinResponse: BroadcastTxCommitResult = await this.wallet?.callMethod(
+            chessRealm,
+            "JoinLobby", // TODO change when API is ready
+            [
+                time.time.toString(),
```

Comment:
I think this is not an object but a basic array.
```
time[0].toString() // time
time[1].toString() // increment
```

Reply:
I've swapped it out for a concrete object type, so we match the calling args of the Realm (time, increment)